### PR TITLE
Complete MNIST train example

### PR DESCRIPTION
**Pull request by Yancey1989**
fixed #94

### CODE REVIEW

File: mnist_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		loss.Backward()
+		batchIdx++
+		if batchIdx%100 == 0 {
+			fmt.Printf("batch: %d, Loss: %s", batchIdx, loss)
+		}
+	}
+	trainLoader.Close()
+	mnist.Close()
+	torch.FinishGC()
+}
```
Comment (wangkuiyi) [inline mnist_test.go:59]: You need a comment here, otherwise, `go test` would not run this example.

```go
// Output:
```

File: mnist_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	trainLoader := torch.NewDataLoader(mnist, 64)
+	opt := torch.SGD(0.1, 0, 0, 0, false)
+	opt.AddParameters(nn.GetParameters(net))
+	batchIdx := 0
+	for trainLoader.Scan() {
+		batch := trainLoader.Batch()
+		pred := net.Forward(batch.Data)
+		loss := F.NllLoss(pred, batch.Target, torch.Tensor{nil}, -100, "mean")
+		opt.ZeroGrad()
+		loss.Backward()
```
Comment (wangkuiyi) [inline mnist_test.go:50]: Currently, this example doesn't converge. I think you will need a line `opt.Step` here.

File: mnist_test.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	}
+	net := NewMNISTNet()
+	transforms := []torch.Transform{torch.NewNormalize(0.1307, 0.3081)}
+	mnist := torch.NewMNIST(dataDir(), transforms)
+	trainLoader := torch.NewDataLoader(mnist, 64)
+	opt := torch.SGD(0.1, 0, 0, 0, false)
+	opt.AddParameters(nn.GetParameters(net))
+	batchIdx := 0
+	for trainLoader.Scan() {
+		batch := trainLoader.Batch()
```
Comment (wangkuiyi) [inline mnist_test.go:46]: Why there is no `torch.GC()` call in this loop?

File: mnist_test.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	torch "github.com/wangkuiyi/gotorch"
+	nn "github.com/wangkuiyi/gotorch/nn"
+	F "github.com/wangkuiyi/gotorch/nn/functional"
+)
+
+type MultiLayerMNISTNet struct {
+	FC1, FC2, FC3 nn.Module
+}
+
+func (n *MultiLayerMNISTNet) Forward(x torch.Tensor) torch.Tensor {
```
Comment (QiJune) [inline mnist_test.go:16]: Is the example here refer to a Python example? Could you please give the link?

File: mnist_test.go (hunk 2 above)
Comment (Yancey1989) [reply mnist_test.go]: fixed, and updated the active function to `Tanh`. Now it converged as the following figure.
image

File: mnist_test.go [hunk 5]
```diff
@@ -16,9 +15,10 @@ type MultiLayerMNISTNet struct {
 func (n *MultiLayerMNISTNet) Forward(x torch.Tensor) torch.Tensor {
 	x = torch.View(x, []int{-1, 28 * 28})
 	x = n.FC1.Forward(x)
-	x = torch.Relu(x)
+	//x = torch.Relu(x)
```
Comment (wangkuiyi) [inline mnist_test.go:18]: Remove comments?

File: mnist_test.go (hunk 2 above)
Comment (Yancey1989) [reply mnist_test.go]: I will port the [item](https://pytorch.org/docs/stable/tensors.html#torch.Tensor.item) function in Go so that we can assert the loss value in the unit test.

File: mnist_test.go (hunk 3 above)
Comment (Yancey1989) [reply mnist_test.go]: It's hidden in `dataloader.Scan()`, c.f. https://github.com/wangkuiyi/gotorch/pull/71#discussion_r466130145
